Gems/RecastNavigation/Code/Source/RecastNavigationSystemComponent.h

```diff
@@ -0,0 +1,47 @@
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <RecastNavigation/RecastNavigationBus.h>
+
+namespace RecastNavigation
+{
+    class RecastNavigationSystemComponent
+        : public AZ::Component
+        , protected RecastNavigationRequestBus::Handler
+        , public AZ::TickBus::Handler
+    {
+    public:
+        AZ_COMPONENT(RecastNavigationSystemComponent, "{9d7ae509-b1db-4889-98bb-941a3f672ca3}");
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
+        RecastNavigationSystemComponent();
+        ~RecastNavigationSystemComponent();
```

Comment:
Nit: should this be virtual destructor? Assume that we may delete components via reference to base class.

Reply:
Override added.